### ISSUE TITLE
The UI becomes very sluggish after decoding many packets and saving to file is enabled

### ISSUE THREAD

**Issue by invader-zimm** [cropinghigh/qstdcdec#2]
To reproduce:
- in the "Packets" tab, enable "Save packets to"
- decode 2000+ packets

Result: the UI becomes very slow to respond. In my case, it starts to happen after decoding 4000-4500 packets. If "Save packets to" is disabled, the UI doesn't slow down (or at least it's not very noticeable).

ScytaleC has some checkboxes which control which packet types to decode and which to ignore. Could a similar feature be added to qstdcdec?

And on the same note, I think that after decoding thousand of messages and saving to file is enabled, the same problem would occur. Which leads me to a feature request: could you please add a way to filter out or discard messages containing certain strings?
For example, I left the decoder running for half a day and the overwhelming majority of decoded messages look like this
````
abcdefghijklmnopqrstuvwxyz1234567890ABCDEFGHIJKLMNOPQRSTUVWXYZ-!
````
I'm assuming this is some type of test message and I would really like to delete/discard them from the decoded messages list (and obviously from the saved json file).

**Comment by cropinghigh** [comment]
Yes, also a known problem. I'm just too lazy and busy to fix it all

**Comment by invader-zimm** [comment]
I'm guessing it has something to do with opening,reading then writing to/from the json file every time a packet is decoded, right? Maybe it would be better to write to file at some intervals, to avoid HDD wear. The workaround I'm using is to enable the write to file checkbox only just before closing down the program, that way the json file is written only once.

What about filtering out certain decoded messages? Would it be difficult to implement?
